_posts/2020-08-15-Named-Graphs.md

```diff
@@ -0,0 +1,391 @@
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a
+where {
+    ?a irishRel:nomName "Find"
+}
+```
+
+In GraphDB, this query will return 105 results.  Why does this return
+so many results?  This is due to the way in which GraphDB constructs
+its [default
+graph](http://graphdb.ontotext.com/documentation/standard/query-behaviour.html):
+
+> > GraphDB constructs the default dataset as follows:
+>
+>     The dataset’s default graph contains the merge of the database’s default graph AND all the database named graphs;
+>
+>     The dataset contains all named graphs from the database.
+
+The reason given in the documentation is thus:
+
+> There are two reasons for this behavior:
+>
+>     It provides an easy way to execute a triple pattern query over all stored RDF statements.
+>
+>     It allows all named graph names to be discovered, i.e., with this query: SELECT ?g { GRAPH ?g { ?s ?p ?o } }.
+
+If the user runs the query:
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+prefix onto: <http://www.ontotext.com/>
+
+select ?a
+from <http://example.com/LL>
+from onto:explicit
+where {
+    ?a irishRel:nomName "Find"
+}
+```
+
+There will be 19 results as from Stardog.  What is `onto:explicit`?
+This has to do with the way in which GraphDB deals with `owl:sameAs`
+equivalency and acts much like Stardog's merging of those nodes that
+have an effect on the result set.  To break this, `onto:explicit`
+pseudo-name graph is used (see [Optimization of
+owl:sameAs](http://graphdb.ontotext.com/documentation/standard/sameas-optimisation.html)).
+
+To return to the query above which contains two named graphs, if the
+user wished to have all the instances of `irishRel:nomName "Find"`
+from LL and Rawl B502 without the `owl:sameAs` equivalencies
+interfering with the results, the query would be:
+
+```sparql
+prefix onto: <http://www.ontotext.com/>
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a ?g
+from <http://example.com/Rawl_B502>
+from named <http://example.com/LL>
+from onto:explicit
+from named onto:explicit
+where {
+    { ?a irishRel:nomName "Find" }
+    union {
+		graph ?g {
+			?a irishRel:nomName "Find"
+		}
+    }	
+}
+```
+
+This returns 44 results.  Although, it would be easier to do the
+following:
+
+```sparql
+prefix onto: <http://www.ontotext.com/>
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a ?g
+from named <http://example.com/Rawl_B502>
+from named <http://example.com/LL>
+from named onto:explicit
+where {
+	graph ?g {
+		?a irishRel:nomName "Find"
+	}
+		
+}
+```
+
+The above is easier to write and read.
+
+# Conclusion
+
+Named graphs allow IrishGen to partition the various MSS sources into
+their own sub-graphs.  This allows the user to interrogate one MS
+tradition or another, which can be important to certain
+investigations.  However, choosing a Triplestore has consequences for
```

Comment:
"can be important to certain investigations" > "is fundamental to serious study of the genealogical tradition"

If you know of any types of investigation where it doesn't really matter what primary sources the information is coming from, feel free to detail those as well...